x-pack/plugins/saved_objects_tagging/public/ui_api/parse_search_query.ts

```diff
@@ -20,13 +20,25 @@ export const buildParseSearchQuery = ({
   cache,
 }: BuildParseSearchQueryOptions): SavedObjectsTaggingApiUi['parseSearchQuery'] => {
   return (query: string, { tagField = 'tag', useName = true }: ParseSearchQueryOptions = {}) => {
-    const parsed = Query.parse(query);
+    let parsed: Query;
+
+    try {
+      parsed = Query.parse(query);
+    } catch (e) {
+      return {
+        searchTerm: query,
+        tagReferences: [],
+        valid: false,
```

Comment:
where we use it?

Reply:
> where we use it?

it's currently unused, just figured it would make sense to have the info as we changed the method's behavior if a future consumer needs to check the success.

> do you want to return error details for the debugging purposes?

I did initially, added an `error?: string` that was the query validation error, and then I removed it. Should I put it back?

Reply:
> Should I put it back?

only if it's used somewhere. rendering an error about wrong format, for example.

Reply:
Not atm. Might in the future if apps want to alert their users about that, but we'll add it then